packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.spec.ts

```diff
@@ -166,5 +166,61 @@ describe("XYChartComponent", () => {
             });
             expect(spy).toHaveBeenCalledWith({ payload: timeframe });
         });
+
+        it("should return true if the series is set to interactive", () => {
+            component.configuration = { interaction: "series" } as ITimeseriesWidgetConfig;
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
+
+        it("should return true if the dataSeries has 'link' populated", () => {
+            component.widgetData = { series: [{} as ITimeseriesWidgetData] };
+            component.widgetData.series[0] = {
+                id: "id",
+                name: "name",
+                description: "description",
+                link: "www.link.com",
+                data: {} as ITimeseriesWidgetSeriesData[],
+            }
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
+
+        it("should return true if the dataSeries has 'secondaryLink' populated", () => {
+            component.widgetData = { series: [{} as ITimeseriesWidgetData] };
+            component.widgetData.series[0] = {
+                id: "id",
+                name: "name",
+                description: "description",
+                secondaryLink: "www.link.com",
+                data: {} as ITimeseriesWidgetSeriesData[],
+            }
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
```

Comment:
Do you mind putting these three tests in their own describe block for `isSeriesInteractive` outside of the `ngOnChanges` describe block.